src/components/standalone/firewall/conntrack/DeleteConntrackRecordModal.vue

```diff
@@ -0,0 +1,117 @@
+<!--
+  Copyright (C) 2024 Nethesis S.r.l.
+  SPDX-License-Identifier: GPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import { ubusCall } from '@/lib/standalone/ubus'
+import { useNotificationsStore } from '@/stores/notifications'
+import { NeInlineNotification, getAxiosErrorMessage } from '@nethesis/vue-components'
+import { NeModal } from '@nethesis/vue-components'
+import { ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import type { ConntrackRecord } from './ConntrackContent.vue'
+
+const { t } = useI18n()
+const notificationsStore = useNotificationsStore()
+const props = defineProps<{
+  visible: boolean
+  itemToDelete?: ConntrackRecord
+}>()
+
+const emit = defineEmits(['close', 'record-deleted'])
+
+const error = ref({
+  notificationDescription: '',
+  notificationDetails: ''
+})
+const isDeleting = ref(false)
+
+async function deleteConntrackRecord() {
+  error.value = {
+    notificationDescription: '',
+    notificationDetails: ''
+  }
+  isDeleting.value = true
+  try {
+    if (props.itemToDelete !== undefined) {
+      await ubusCall('ns.conntrack', 'drop', {
+        id: props.itemToDelete.id
+      })
+      emit('record-deleted')
+
+      // show toast notification
+      setTimeout(() => {
+        notificationsStore.createNotification({
+          title: t('standalone.conntrack.conntrack_record_removed'),
+          kind: 'success'
+        })
+      }, 500)
+    } else {
+      await ubusCall('ns.conntrack', 'drop_all', {})
+      emit('record-deleted')
+
+      // show toast notification
+      setTimeout(() => {
+        notificationsStore.createNotification({
+          title: t('standalone.conntrack.conntrack_record_all_removed'),
+          kind: 'success'
+        })
+      }, 500)
```

Comment:
Notifications should be called outside the modal, if for any reason Vue unloads the component while hidden the code won't run

Reply:
I am not sure what I did is good but you can find this way to use the toaster elsewhere in a modal

https://github.com/search?q=repo%3ANethServer%2Fnethsecurity-ui%20createNotification&type=code


if we follow my idea we can have the plural of the translation -> connection deleted successfully  vs connections deleted successfully

with your idea we have only one translation

Reply:
Why...?
On successfully deleted event, before closing the modal, you can check how `itemToDelete` is set and set a title accordingly

I understand that there are some places where the same behaviour is present, however every component should mind its own business to keep features enclosed. Since `ConntrackContent.vue` allows the deletion (otherwise no button will be shown to do so) it should even take care of everything else regarding what happens inside the component (for instance, reloading the data once the modal is dismissed with a delete, which already does that)

Reply:
done :)